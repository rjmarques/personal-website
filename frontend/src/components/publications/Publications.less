@import "../../theme.less";

.publications {
  .publications-entry {
    padding-bottom: @card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
