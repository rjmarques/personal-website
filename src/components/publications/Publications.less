@import "../../theme.less";

.Publications {
  .Publications-entry {
    padding-bottom: @card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
