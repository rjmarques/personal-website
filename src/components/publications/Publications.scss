@use "../../theme.scss" as *;

.publications {
  .publications-entry {
    padding-bottom: $card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
