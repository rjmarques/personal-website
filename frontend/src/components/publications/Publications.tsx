import React from "react";

import ContentSection from "../common/ContentSection";

import "./Publications.less";

interface IPublication {
  conference: string;
  paper: string;
  urls: string[];
}

const publications: IPublication[] = [
  {
    conference: "Sac'14",
    paper:
      "On the Support of Task-Parallel Algorithmic Skeletons for Multi-GPU Computing",
    urls: ["https://dl.acm.org/doi/10.1145/2554850.2555018"],
  },
  {
    conference: "Euro-Par 2013",
    paper:
      "Algorithmic Skeleton Framework for the Orchestration of GPU Computations",
    urls: ["http://link.springer.com/chapter/10.1007%2F978-3-642-40047-6_86"],
  },
];

const Publications = () => (
  <ContentSection
    title="Publications"
    subTitle="Papers I have published"
    wrapperClass="publications"
  >
    <ul className="materialize">
      {publications.map((pub) => (
        <li key={pub.conference} className="publications-entry">
          <div className="app-card-wide">
            <h3>{pub.conference}</h3>
            <div className="app-card-subtitle">
              <span>{pub.paper}</span>
            </div>
            {pub.urls.map((url) => (
              <div key={url}>
                <a href={url} target="_blank" rel="noopener noreferrer">
                  {url}
                </a>
              </div>
            ))}
          </div>
        </li>
      ))}
    </ul>
  </ContentSection>
);

export default Publications;
