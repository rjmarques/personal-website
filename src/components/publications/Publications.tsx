import * as React from "react";

import SectionTitle from "../common/SectionTitle";

import "./Publications.less";

interface IPublication {
  conference: string;
  paper: string;
  urls: string[];
}

class Publications extends React.Component {
  private publications: IPublication[] = [
    {
      conference: "Sac'14",
      paper:
        "On the Support of Task-Parallel Algorithmic Skeletons for Multi-GPU Computing",
      urls: ["http://nova-lincs.di.fct.unl.pt/publication/1329"]
    },
    {
      conference: "Euro-Par 2013",
      paper:
        "Algorithmic Skeleton Framework for the Orchestration of GPU Computations",
      urls: ["http://link.springer.com/chapter/10.1007%2F978-3-642-40047-6_86"]
    }
  ];

  public render() {
    return (
      <section className="Publications">
        <div className="content">
          <SectionTitle
            title="Publications"
            subTitle="Papers I have published"
          />
          <div>{this.getPublications()}</div>
        </div>
      </section>
    );
  }

  private getPublications(): JSX.Element {
    return (
      <ul>
        {this.publications.map(pub => (
          <li key={pub.conference} className="Publications-entry">
            <div className="App-card-wide">
              <h3>{pub.conference}</h3>
              <div className="App-card-subtitle">
                <span>{pub.paper}</span>
              </div>
              {pub.urls.map(url => (
                <div key={url}>
                  <a href={url} target="_blank">
                    {url}
                  </a>
                </div>
              ))}
            </div>
          </li>
        ))}
      </ul>
    );
  }
}

export default Publications;
