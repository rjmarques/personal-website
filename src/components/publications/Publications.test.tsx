import React from "react";
import renderer from "react-test-renderer";

import Publications from "./Publications";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const component = renderer.create(<Publications />);

    const tree = component.toJSON();
    expect(component).toMatchSnapshot();
  });
});
