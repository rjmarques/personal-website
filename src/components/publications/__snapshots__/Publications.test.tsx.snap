// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="Publications"
>
  <div
    className="content"
  >
    <div
      className="Content-title"
    >
      <h2>
        Publications
      </h2>
      <span
        className="Section-sub-title"
      >
        Papers I have published
      </span>
    </div>
    <ul
      className="materialize"
    >
      <li
        className="Publications-entry"
      >
        <div
          className="App-card-wide"
        >
          <h3>
            Sac'14
          </h3>
          <div
            className="App-card-subtitle"
          >
            <span>
              On the Support of Task-Parallel Algorithmic Skeletons for Multi-GPU Computing
            </span>
          </div>
          <div>
            <a
              href="http://nova-lincs.di.fct.unl.pt/publication/1329"
              rel="noopener noreferrer"
              target="_blank"
            >
              http://nova-lincs.di.fct.unl.pt/publication/1329
            </a>
          </div>
        </div>
      </li>
      <li
        className="Publications-entry"
      >
        <div
          className="App-card-wide"
        >
          <h3>
            Euro-Par 2013
          </h3>
          <div
            className="App-card-subtitle"
          >
            <span>
              Algorithmic Skeleton Framework for the Orchestration of GPU Computations
            </span>
          </div>
          <div>
            <a
              href="http://link.springer.com/chapter/10.1007%2F978-3-642-40047-6_86"
              rel="noopener noreferrer"
              target="_blank"
            >
              http://link.springer.com/chapter/10.1007%2F978-3-642-40047-6_86
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</section>
`;
